source/MapPlanetCard.h

```diff
@@ -0,0 +1,103 @@
+/* MapPlanet.cpp
+Copyright (c) 2022 by Hurleveur
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef MAP_PLANET_CARD_H_
+#define MAP_PLANET_CARD_H_
+
+#include "MapPanel.h"
+#include "Sprite.h"
+
+#include <vector>
+#include <string>
+
+class Point;
+class StellarObject;
+
+
+
+// Used to manage the display of a single planet in the MapDetailPanel.
+class MapPlanetCard {
+public:
+	enum class ClickAction : int {
+		SHOW_REPUTATION = MapPanel::SHOW_REPUTATION,
+		SHOW_SHIPYARD = MapPanel::SHOW_SHIPYARD,
+		SHOW_OUTFITTER = MapPanel::SHOW_OUTFITTER,
+		SHOW_VISITED = MapPanel::SHOW_VISITED,
+		GOTO_SHIPYARD = 3,
+		GOTO_OUTFITTER = 2,
+		SELECTED = 1,
+		NONE = 0
+	};
+
+
+public:
+	// For the orbit selection to work properly this has to be a planet.
+	explicit MapPlanetCard(const StellarObject &object, bool hasVisited);
+	~MapPlanetCard();
+	// Return if this one was clicked, whether or not we did something about it.
+	ClickAction Click(int x, int y, int clicks);
+	// Draw this at the corresponding scoll; if it is not outside bounds, and return if we drew it.
+	bool DrawIfFits(const Point &uiPoint);
+	// If this object is currently being shown.
+	bool Shown() const;
+	// Return the space available for this planet card on its current position.
+	double AvailableSpace() const;
+
+	const Planet *getPlanet() const;
+
+
+public:
+	static void setScroll(double newScroll);
+	static double getScroll();
+	static void clear();
```

Comment:
Well that's what it was indeed but MapDetailPanel is already so charged
I see what you have in mind, this would cause problems if we had two MapDetailPanels, I'll change it then, but this class accesses it more than said MapDetailPanel, each element looks at it to know where to be drawn

Reply:
I'll need to make it static though to allow the MapPlanetCards to access it